src/python/pants/build_graph/target.py

```diff
@@ -113,17 +112,10 @@ def check_unknown(self, target, kwargs, payload):
       """
       ignore_params = set((self.get_options().ignored or {}).get(target.type_alias, ()))
       unknown_args = {arg: value for arg, value in kwargs.items() if arg not in ignore_params}
-      if 'source' in unknown_args and 'sources' in payload.as_dict():
-        unknown_args.pop('source')
-      if 'sources' in unknown_args:
-        if 'sources' in payload.as_dict():
-          deprecated_conditional(
-            lambda: True,
-            '1.11.0.dev0',
-            ('The source argument is deprecated - it gets automatically promoted to sources.'
-             'Target {} should just use a sources argument. No BUILD files need changing. '
-             'The source argument will stop being populated -').format(target.type_alias),
-          )
+      if 'sources' in payload.as_dict():
```

Comment:
I noticed this was last touched in https://github.com/pantsbuild/pants/commit/450de702414f87f563081ddefaefd8a554de07a3 - it feels like we should be able to drop these checks / pops and allow an error to be raised if they're included (or move them to `ignored_args`?), @illicitonion @cosmicexplorer might ya'll know on if this can be removed, moved to `ignored_args`, or should be left as-is?

Reply:
Yeah, I think you should be able to remove this entire block, and just treat `sources` like any other arg :) Good spot!

Reply:
Getting rid of this caused [quite a few test failures](https://travis-ci.org/pantsbuild/pants/builds/503824443), most (maybe all? from what I've spot checked so far at least) of which are from having `source=` within `BUILD` files across pants. Would it be safe to assume those should be updated to `sources=[...]` accordingly, or might it be preferable to revert https://github.com/pantsbuild/pants/pull/7332/commits/c09f9cbab4a14477db1b14a7ac0295655fe36893 for now?

Reply:
Oh dear - please revert that change and file a ticket assigned to me :) Sorry about that!

Reply:
No worries! Will revert and get a ticket open for that.

Reply:
Consider adding a TODO comment linking to 7357 here.

Reply:
[Done!](https://github.com/pantsbuild/pants/pull/7332/commits/ea02d63d2ef505a75ec224a55a81989c95a75e2e)